### ISSUE TITLE
Points not decreasing

### ISSUE THREAD

**Issue by tre-dev** [animir/node-rate-limiter-flexible#100]
I'm using `nore-rate-limiter-flexible` via example memory code within my stack to check specific requests, but the remaining points are never decreasing.  (even if I spam multiple per second)

Code:
```
const rate = new RateLimiterMemory({points: 5, duration: 1})

rate
.consume(req.ip, 2)
.then((e) => {
    console.log('ip', req.ip)
    console.log('response', e)
})
.catch(() => {
  console.error('rate limit')
})
```

Response:
```
ip: ::1
response: {
  remainingPoints: 3,
  msBeforeNext: 1000,
  consumedPoints: 2,
  isFirstInDuration: true
}
```

**Comment by animir** [comment]
@tre-dev hi, you set 5 points, then consumed 2. `remainingPoints` equals to 3. It seems fare to me. How do you spam multiple times per second?

**Comment by tre-dev** [comment]
I only pasted a single event. Let's say I trigger an API request via a button click, if I click on that button 4 or 5 times per second,  it should trigger the rate limit, correct? Because that's what I did and the points always stayed the same as the one that I've pasted as a response.

**Comment by animir** [comment]
@tre-dev First of all, you use memory limiter. How do you run your application? Is it cluster or single nodejs process?
Secondly, I wouldn't be so sure, you can test this with button: even if you are able to click it 4 times per second, there is no guarantee all 4 requests would go within single second period. They could reach server during 2 seconds.

**Comment by tre-dev** [comment]
Yes, I'm using memory limiter. Planning to use central store later on, but so far it's fine. I haven't tested in staging/prod, only in dev environment within a single nodejs process. I've added the rate limiter within a single graphql resolver. Can I somehow see a list of recent points deducations? Or is there another way that it's saving & loading it correctly to/from memory?
Thanks!

**Comment by animir** [comment]
@tre-dev You can try to consume points programmatically, e.g. call consume several times and log resolved or rejected result for the same key. You can also get internal memory storage with `rate._memoryStorage._storage` and log it. Then share results here, we'll figure it out.

**Comment by tre-dev** [comment]
Okay, I think I got the problem. I recreated the RateLimiter every request to be able to set dynamic options, which resets the memory. I'm setting different versions now once on startup. Thanks for the help!

**Comment by ShekharBhabad** [comment]
Hi, 
I am also facing the same issue when we are trying to use memory code instead of the share memory (redis/memcached)
points are not getting decreased even when I am hitting a particular endpoint using ab testing.

@tre-dev  what do you meant when you said -> "I recreated the RateLimiter every request to be able to set dynamic options, which resets the memory. I'm setting different versions now once on startup"

what changes will I have to make ?

I am always getting this in response
 {
  remainingPoints: 50,
  msBeforeNext: 1000,
  consumedPoints: 50,
  isFirstInDuration: true
}

so basically its treating all the requests as first requests (isFirstInDuration is always true) and never decreasing points after first time.
